modules/microshift-downloading-container-images.adoc

```diff
@@ -43,23 +43,90 @@ $ IMAGE_LIST_FILE=~/microshift-container-refs.txt
 $ IMAGE_LOCAL_DIR=~/microshift-containers
 ----
 
-. Run the following script to download the container images to the `${IMAGE_LOCAL_DIR}` directory:
+. Run one of the following scripts that matches your use case:
+
+.. Run the following script to pull the images from ${IMAGE_LOCAL_DIR} to the mirror registry:
 +
 [source,terminal]
 ----
-while read -r src_img ; do
-   # Remove the source registry prefix
-   dst_img=$(echo "${src_img}" | cut -d '/' -f 2-)
+pushd "${IMAGE_LOCAL_DIR}" >/dev/null
+while read -r src_manifest ; do
+  local src_img
+  src_img=$(dirname "${src_manifest}")
+  # Add the target registry prefix and remove SHA
+  local -r dst_img="${TARGET_REGISTRY}/${src_img}"
+  local -r dst_img_no_tag="${TARGET_REGISTRY}/${src_img%%[@:]*}"
+  # Run the image upload and tag commands
+  echo "Uploading '${src_img}' to '${dst_img}'"
+  skopeo copy --all --quiet \
+     --preserve-digests \
+     --authfile "${IMAGE_PULL_FILE}" \
+     dir://"${local_dir}/${src_img}" docker://"${dst_img}"
+  echo "Tagging '${dst_img}' as '${dst_img_no_tag}:latest'"
+  skopeo copy --all --quiet \
+    --preserve-digests \
+    --authfile "${IMAGE_PULL_FILE}" \
+    docker://"${dst_img}" docker://"${dst_img_no_tag}:latest"
+done < <(find . -type f -name manifest.json -printf '%P\n')
+----
 
-   # Run the image download command
-   echo "Downloading '${src_img}' to '${IMAGE_LOCAL_DIR}'"
-   mkdir -p "${IMAGE_LOCAL_DIR}/${dst_img}"
-   skopeo copy --all --quiet \
-      --preserve-digests \
-      --authfile "${PULL_SECRET_FILE}" \
-      docker://"${src_img}" dir://"${IMAGE_LOCAL_DIR}/${dst_img}"
+.. Run the following script to pull the images directly from the local registry to the mirror registry, skipping the download of data in a destination directory:
++
+[source,terminal]
+----
+while read -r src_img; do
+    # Remove the source registry prefix and SHA
+    local dst_img
+    dst_img=$(echo "${src_img}" | cut -d '/' -f 2-)
+    local dst_img_no_tag
+    dst_img_no_tag="${dst_img%%[@:]*}"
+    # Add the target registry prefix
+    dst_img="${TARGET_REGISTRY}/${dst_img}"
+    dst_img_no_tag="${TARGET_REGISTRY}/${dst_img_no_tag}"
+    # Mirror the image and tag it in the target registry
+    echo "Mirroring '${src_img}' to '${dst_img}'"
+    skopeo copy --all --quiet \
+        --preserve-digests \
+        --authfile "${PULL_SECRET_FILE}" \
+        docker://"${src_img}" docker://"${dst_img}"
 
+    echo "Tagging '${dst_img_no_tag}' as 'latest'"
+    skopeo copy --all --quiet \
+        --preserve-digests \
+        --authfile "${PULL_SECRET_FILE}" \
+        docker://"${dst_img}" docker://"${dst_img_no_tag}:latest"
 done < "${IMAGE_LIST_FILE}"
 ----
++
+[NOTE]
+====
+You can use the following method with disconnected registries when copying the local directory over to an isolated host.
+====
++
+.. Run the following script to pull the images from the remote registry to the local registry, then pull from your local registry to the mirror registry:
```

Comment:
Commands here are the same as in the direct remote registry -> mirror registry? I am missing the pull part to local disk.

Reply:
@pacevedom Can you provide?